dockerfiles/base/requirements.txt

```diff
@@ -1,7 +1,7 @@
 blosc~=1.7
 dask-kubernetes~=0.11.0
 dask-ml~=1.4,<1.9.0
-dask[complete]~=2021.11.2
+dask[complete]~=2023.9.0
```

Comment:
Just for me to clarify - Is there a reason we explicitly require 2023.9.0 and not a range? More generic question - wow we decide what package we require specific version  (or subversion) like dask, and what package we allow wider range - like pandas?

Reply:
Tilde is shorthand notation. `~=2023.9.0` is equivalent to `>=2023.9.0,<2023.10`.

Regarding the second question, this PR preserves the current policy, but as to why this was done in the first place, I can think of good reasoning for it. Pandas a is user library, and our goal should be to support as wide a range as is feasible. Users are very opinionated about which pandas library they use. My understanding is that this is not as much the case with dask. Supporting a wide range means you actually need to verify the whole range for proper testing (except for patch versions which are normally assumed to be "safe").